pkg/query-service/app/inframetrics/jobs.go

```diff
@@ -0,0 +1,498 @@
+package inframetrics
+
+import (
+	"context"
+	"math"
+	"sort"
+
+	"go.signoz.io/signoz/pkg/query-service/app/metrics/v4/helpers"
+	"go.signoz.io/signoz/pkg/query-service/common"
+	"go.signoz.io/signoz/pkg/query-service/interfaces"
+	"go.signoz.io/signoz/pkg/query-service/model"
+	v3 "go.signoz.io/signoz/pkg/query-service/model/v3"
+	"go.signoz.io/signoz/pkg/query-service/postprocess"
+	"golang.org/x/exp/slices"
+)
+
+var (
+	metricToUseForJobs = "k8s_pod_cpu_utilization"
+	k8sJobNameAttrKey  = "k8s_job_name"
+
+	metricNamesForJobs = map[string]string{
+		"desired_successful_pods": "k8s_job_desired_successful_pods",
+		"active_pods":             "k8s_job_active_pods",
+		"failed_pods":             "k8s_job_failed_pods",
+		"successful_pods":         "k8s_job_successful_pods",
+	}
+
+	jobAttrsToEnrich = []string{
+		"k8s_job_name",
+		"k8s_namespace_name",
+		"k8s_cluster_name",
+	}
+
+	queryNamesForJobs = map[string][]string{
+		"cpu":             {"A"},
+		"cpu_request":     {"B", "A"},
+		"cpu_limit":       {"C", "A"},
+		"memory":          {"D"},
+		"memory_request":  {"E", "D"},
+		"memory_limit":    {"F", "D"},
+		"restarts":        {"G", "A"},
+		"desired_pods":    {"H"},
+		"active_pods":     {"I"},
+		"failed_pods":     {"J"},
+		"successful_pods": {"K"},
+	}
```

Comment:
nit: rename a, b, c, to some better query names
eg for kafka_fetch_latency I used "latency"
reason: code readability